### PR TITLE
#5257 Refactor RemoteFolderHandler

### PR DESCRIPTION
**Pull request by chia7712**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #5257)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/test/scala/oharastream/ohara/agent/TestAgent.scala [hunk 1]
```diff
@@ -52,13 +46,14 @@ class TestAgent extends OharaTest {
...
-      case (command, response) =>
-        val agent =
-          Agent.builder.hostname(server.hostname).port(server.port).user(server.user).password(server.password).build
-        try agent.execute(command).get.split("\n") shouldBe response
-        finally agent.close()
+    def assertResponse(request: String, response: java.util.List[String]): Unit = {
+      val agent =
+        Agent.builder.hostname(server.hostname).port(server.port).user(server.user).password(server.password).build
+      try agent.execute(request).get.split("\n").toSeq shouldBe response.asScala.toSeq
+      finally agent.close()
```
Comment (jackyoh) [inline ohara-agent/src/test/scala/oharastream/ohara/agent/TestAgent.scala:53]: 這邊應該可以用 Releasable.close(agent) 吧。

Comment (chia7712) [reply ohara-agent/src/test/scala/oharastream/ohara/agent/TestAgent.scala:53]: 這邊是測試代碼所以讓它拋出錯誤

Comment (jackyoh) [reply ohara-agent/src/test/scala/oharastream/ohara/agent/TestAgent.scala:53]: 有一個疑惑的地方，就是拋錯誤怎麼沒看到類似 an[IllegalArgumentException] thrown by 的語法去捕捉錯誤

Comment (chia7712) [reply ohara-agent/src/test/scala/oharastream/ohara/agent/TestAgent.scala:53]: 我的意思是有錯誤就讓它跑出來而不是吃掉

Comment (jackyoh) [reply ohara-agent/src/test/scala/oharastream/ohara/agent/TestAgent.scala:53]: 喔,了解。 Thank you.

**Pull request merged**